### ISSUE TITLE
Confused about masking(conv mask) in convcap's forward

### ISSUE THREAD

**Issue by joelxiangnanchen** [aditya12agd5/convcap#12]
Hi, Author, thx for great code publication. After read `convcap.py`'s code, I have a question about doing masking in training stage. From paper, we will do mask on conv1d to use just only past info. And the code as follows:
```
for i, conv in enumerate(self.convs):
      
      if(i == 0):
        x = x.transpose(2, 1)
        residual = self.resproj(x)
        residual = residual.transpose(2, 1)
        x = x.transpose(2, 1)
      else:
        residual = x

      x = F.dropout(x, p=self.dropout, training=self.training)

     #do conv here
      x = conv(x)
      x = x[:,:,:-self.pad]

      x = F.glu(x, dim=1)

      if(self.is_attention):
        attn = self.attention[i]
        x = x.transpose(2, 1)
        x, attn_buffer = attn(x, wordemb, imgsfeats)
        x = x.transpose(2, 1)
    
      x = (x+residual)*math.sqrt(.5)
```
I don't understand where done the mask. Under my understanding, the conv operates on whole sequence that conv window does not be masked.  In this block, conv should 'see' the right side's words. Did `x = x[:,:,:-self.pad]` mask the seq by shifting seq to most left? Thus the conv window's most right is the pivot word not like the traditional conv window treating center word as it? 
Thx very much, be grateful for discussion.

**Comment by wangru26** [comment]
I agree that there is this problem. Have you solved it?

**Comment by liyongsheng-tech** [comment]
I also don't understand mask, it seems that there is no code?

**Comment by liyongsheng-tech** [comment]
> I agree that there is this problem. Have you solved it?

The causal convolution used in the code
